src/Microsoft.DocAsCode.Build.RestApi/RestApiHelper.cs

```diff
@@ -27,25 +32,66 @@ public static string FormatDefinitionSinglePath(string reference)
         /// </summary>
         /// <param name="reference"></param>
         /// <returns></returns>
-        public static Tuple<string, string> FormatReferenceFullPath(string reference)
+        public static SwaggerReference FormatReferenceFullPath(string reference)
         {
+            if (reference == null)
+            {
+                throw new ArgumentNullException(nameof(reference));
+            }
+
             // Decode for URI Fragment Identifier Representation
             if (reference.StartsWith("#/"))
             {
                 // Reuse relative path, to decode the values inside '/'.
                 var path = reference.Substring(2);
                 var decodedPath = ((RelativePath)path).UrlDecode();
-                return Tuple.Create("/" + decodedPath, decodedPath.FileName);
+                return new SwaggerReference
+                {
+                    Type = SwaggerReferenceType.InternalReference,
+                    Path = "/" + decodedPath,
+                    Name = decodedPath.FileName
+                };
+            }
+
+            // External reference json
+            if (reference.EndsWith(".json", StringComparison.OrdinalIgnoreCase))
+            {
+                return new SwaggerReference
+                {
+                    Type = SwaggerReferenceType.ExternalReference,
+                    Path = reference,
+                    Name = Path.GetFileNameWithoutExtension(reference)
+                };
             }
 
             // Not decode for JSON String Representation
             if (reference.StartsWith("/"))
             {
                 var fileName = reference.Split('/').Last();
-                return Tuple.Create(reference, fileName);
+                return new SwaggerReference
+                {
+                    Type = SwaggerReferenceType.InternalReference,
+                    Path = reference,
+                    Name = fileName
+                };
             }
 
-            throw new InvalidOperationException($"Full reference path \"{reference}\" must start with '/' or '#/'");
+            throw new InvalidOperationException($"Reference path \"{reference}\" is not supported now");
+        }
+
+        public class SwaggerReference
```

Comment:
yes plan to

---
In reply to: [93162767](https://github.com/dotnet/docfx/pull/1019#discussion_r93162767) [](ancestors = 93162767)